### PR TITLE
[Follow-Up] to #2815: Hidden/private tests

### PR DESCRIPTION
**Pull request by b-fein**
### Checklist
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Description
Follow-up to #2815

- SQL queries are now uppercase (I also reformatted the remaining ones in `ProgrammingExerciseRepository` for consistency and readability)
- Restrain width of dropdowns on ‘Configure Grading’ page (Is there a way to actually wrap the text content?)
Screenshot_20210329_201834

### Steps for Testing
1. Open the ‘Configure Grading’ page and check that dropdowns are no longer cut off.

### Test Coverage
unchanged

### CODE REVIEW

**Review by stefanwaldhauser — DISMISSED** [review]
Looks good :) Was initially confused by the XXX.id IS NULL but this seems to be the case because of the left join

File: src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java [hunk 1]
```diff
@@ -29,9 +29,19 @@
...
+     * @param courseId the course the returned programming exercises belong to.
+     * @return all exercises for the given course with only the latest results for solution and template each (if present).
+     */
+    @Query("""
+            SELECT pe FROM ProgrammingExercise pe LEFT JOIN FETCH pe.templateParticipation tp LEFT JOIN FETCH pe.solutionParticipation sp
+            LEFT JOIN FETCH tp.results AS tpr LEFT JOIN FETCH sp.results AS spr
+            WHERE pe.course.id = :#{#courseId}
+                AND (tpr.id = (SELECT MAX(id) FROM tp.results) OR tpr.id IS NULL)
+                AND (spr.id = (SELECT MAX(id) FROM sp.results) OR spr.id IS NULL)
+            """)
```
Comment (derLalla) [inline src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java:44]: to further increase the readability we often put each join into a new line

```suggestion
    @Query("""
            SELECT pe FROM ProgrammingExercise pe
            LEFT JOIN FETCH pe.templateParticipation tp
            LEFT JOIN FETCH pe.solutionParticipation sp
            LEFT JOIN FETCH tp.results tpr
            LEFT JOIN FETCH sp.results spr
            WHERE pe.course.id = :#{#courseId}
                AND (tpr.id = (SELECT MAX(id) FROM tp.results) OR tpr.id IS NULL)
                AND (spr.id = (SELECT MAX(id) FROM sp.results) OR spr.id IS NULL)
            """)
```

File: src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java [hunk 2]
```diff
@@ -92,14 +114,31 @@
...
+                    SELECT coursePe.id
+                    FROM ProgrammingExercise coursePe
+                    WHERE coursePe.course.instructorGroupName IN :groups AND (coursePe.title LIKE %:partialTitle% OR coursePe.course.title LIKE %:partialCourseTitle%)
+                ) OR pe.id IN (
+                    SELECT examPe.id
+                    FROM ProgrammingExercise examPe
+                    WHERE examPe.exerciseGroup.exam.course.instructorGroupName IN :groups AND (examPe.title LIKE %:partialTitle% OR examPe.exerciseGroup.exam.course.title LIKE %:partialCourseTitle%)
+                )
+            )
+            """)
```
Comment (derLalla) [inline src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java:130]:
```suggestion
    @Query("""
            SELECT pe FROM ProgrammingExercise pe
            WHERE pe.shortName IS NOT NULL AND (
                pe.id IN (
                    SELECT coursePe.id
                    FROM ProgrammingExercise coursePe
                    WHERE coursePe.course.instructorGroupName IN :groups
                        AND (coursePe.title LIKE %:partialTitle% OR coursePe.course.title LIKE %:partialCourseTitle%)
                ) OR pe.id IN (
                    SELECT examPe.id
                    FROM ProgrammingExercise examPe
                    WHERE examPe.exerciseGroup.exam.course.instructorGroupName IN :groups
                        AND (examPe.title LIKE %:partialTitle% OR examPe.exerciseGroup.exam.course.title LIKE %:partialCourseTitle%)
                )
            )
            """)
```

**Review by derLalla — CHANGES_REQUESTED** [review]
it was working fine in the testing session. I have left some remarks in the code.

**Review by krusche — APPROVED** [review]

**Pull request merged**